### PR TITLE
Remove one-off dynamic invocations

### PR DESCRIPTION
**Pull request by srawlins**
This PR is a smattering of files that each need 1 or 2 dynamic invocation fixes. All pretty simple.

Work towards https://github.com/flutter/devtools/issues/6929

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or there is a reason for not adding tests.

![build.yaml badge]

If you need help, consider asking for help on [Discord].

[Contributor Guide]: https://github.com/flutter/devtools/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[build.yaml badge]: https://github.com/flutter/devtools/actions/workflows/build.yaml/badge.svg

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/panes/chart/chart_pane.dart [hunk 1]
```diff
@@ -590,7 +590,7 @@ class _MemoryChartPaneState extends State<MemoryChartPane>
     } else if (event[eventName] == devToolsEvent &&
         event.containsKey(customEvent)) {
       final custom = event[customEvent] as Map<dynamic, dynamic>;
-      final data = custom[customEventData];
+      final data = custom[customEventData] as Map<dynamic, dynamic>;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/chart/chart_pane.dart:593]: should we also be switching `dynamic` out here and above for `Object?`

File: packages/devtools_app/lib/src/screens/profiler/cpu_profile_service.dart [hunk 2]
```diff
@@ -49,7 +49,8 @@ extension CpuProfilerExtension on VmService {
       for (int i = 0; i < samples.length; ++i) {
         final cpuSample = samples[i];
         final rawSample = rawSamples[i];
-        cpuSample.setCodeStack(rawSample[kCodeStack].cast<int>());
+        final codeStack = rawSample[kCodeStack] as List;
+        cpuSample.setCodeStack(codeStack.cast<int>());
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/profiler/cpu_profile_service.dart:53]: consider casting above so we can just pass a var into setCodeStack

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_app/lib/src/screens/profiler/cpu_profile_service.dart (hunk 2 above)
Comment (srawlins) [reply packages/devtools_app/lib/src/screens/profiler/cpu_profile_service.dart]: Done

File: packages/devtools_app/lib/src/screens/memory/panes/chart/chart_pane.dart (hunk 1 above)
Comment (srawlins) [reply packages/devtools_app/lib/src/screens/memory/panes/chart/chart_pane.dart]: Good catch, done.

**Pull request merged**